### PR TITLE
codeintel: expose tags associated with upload's commit

### PR DESCRIPTION
**Pull request by Strum355**
Adds an `outlineSecondary` variant Badge containing the tag(s) associated with the commit an upload is associated with. Displayed in the uploads list and upload page as shown below:

image

image

## Test plan

Unit tests, storybook and manual local testing done

### CODE REVIEW

File: client/web/src/enterprise/codeintel/shared/components/CodeIntelUploadOrIndexCommitTags.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            <>
+                tagged as{' '}
+                {tags
+                    .slice(0, 3)
+                    .map<React.ReactNode>(tag => (
+                        <Badge key={tag} variant="outlineSecondary">
+                            {tag}
+                        </Badge>
+                    ))
+                    .reduce((previous, current) => [previous, ', ', current])}
```
Comment (Numbers88s) [inline client/web/src/enterprise/codeintel/shared/components/CodeIntelUploadOrIndexCommitTags.tsx:23]: Hey @Strum355, trying to understand this one. Wanted to give a heads up before it's merged in. You might need to spread `previous` but don't fully understand what you'd like the output to be. 

I'm pulling the repo down to test it but just wanted to give a heads up

Comment (Strum355) [reply client/web/src/enterprise/codeintel/shared/components/CodeIntelUploadOrIndexCommitTags.tsx:23]: This seems to be "the way" to join react components, in this case joined on `, ` text. SO link: https://stackoverflow.com/questions/34034038/how-to-render-react-components-by-using-map-and-join
image

Comment (Numbers88s) [reply client/web/src/enterprise/codeintel/shared/components/CodeIntelUploadOrIndexCommitTags.tsx:23]: > React is perfectly happy with nested arrays.

Thanks for posting the SO. The quote above was what I was worried about. That it wouldn't be okay with a nested array cause that's what you're doing without the spread operator. 

Anyway 👍

File: client/web/src/enterprise/codeintel/uploads/components/CodeIntelUploadNode.tsx [hunk 2]
```diff
@@ -37,7 +38,11 @@ export const CodeIntelUploadNode: FunctionComponent<React.PropsWithChildren<Code
             <div>
                 <span className="mr-2 d-block d-mdinline-block">
                     Directory <CodeIntelUploadOrIndexRoot node={node} /> indexed at commit{' '}
-                    <CodeIntelUploadOrIndexCommit node={node} /> by <CodeIntelUploadOrIndexIndexer node={node} />
+                    <CodeIntelUploadOrIndexCommit node={node} />
+                    {', '}
+                    <CodeIntelUploadOrIndexCommitTags tags={node.tags} />
+                    {', '}
```
Comment (efritz) [inline client/web/src/enterprise/codeintel/uploads/components/CodeIntelUploadNode.tsx:44]: Won't this print `,` twice if there are no tags?

**Review by efritz — APPROVED** [review]

File: client/web/src/enterprise/codeintel/uploads/components/CodeIntelUploadNode.tsx (hunk 2 above)
Comment (Strum355) [reply client/web/src/enterprise/codeintel/uploads/components/CodeIntelUploadNode.tsx:44]: :simple_smile: yes

Comment (efritz) [reply client/web/src/enterprise/codeintel/uploads/components/CodeIntelUploadNode.tsx:44]: It's in the screenshot you posted too :laff:

Comment (Strum355) [reply client/web/src/enterprise/codeintel/uploads/components/CodeIntelUploadNode.tsx:44]: Fixed

**Comment by rrhyne** [comment]
@Strum355 we have an outline badge defined in the design system that would work for your purposes. I'm surprised that it is not already in the storybook badge components. Can someone from @sourcegraph/frontend-platform review this and tell us where the discrepancy is? 

Here is where design's outline badge lives in WDS figma file:
https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/%E2%9C%B3%EF%B8%8F-Wildcard-Design-System?node-id=3654%3A50980

Also note that batch changes represent branches with the secondary badge. Ex: [here](https://demo.sourcegraph.com/users/dan.diemer/batch-changes/upgrade-log4j-v2.16-gradle?visible=3) I like your implementation better, but we would need to change how batch does this to maintain consistency. cc @eseliger.

**Comment by rrhyne** [comment]
Note that while quickly researching this, I may have found a few instances of an outline badge being added into specific features (but not the design system) to add the element locally. 

image

image

**Pull request merged**